unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/States/EditorMode/FreeCameraMovement.cs

```diff
@@ -0,0 +1,230 @@
+    [SerializeField] internal InputAction_Hold advanceLeftInputAction;
+    [SerializeField] internal InputAction_Hold advanceRightInputAction;
+    [SerializeField] internal InputAction_Hold advanceUpInputAction;
+    [SerializeField] internal InputAction_Hold advanceDownInputAction;
+
+
+    private float yaw = 0f;
+    private float pitch = 0f;
+
+    bool isCameraAbleToMove = true, isAdvancingFoward = false,isAdvancingBackward = false, isAdvancingLeft = false, isAdvancingRight = false, isAdvancingUp = false, isAdvancingDown = false;
+
+    Coroutine smoothLookAtCor;
+    private void Awake()
+    {
+        builderInputWrapper.OnMouseDrag += MouseDrag;
+        builderInputWrapper.OnMouseDragRaw += MouseDragRaw;
+        builderInputWrapper.OnMouseWheel += MouseWheel;
+
+        DCLBuilderGizmoManager.OnGizmoTransformObjectStart += OnGizmoTransformObjectStart;
+        DCLBuilderGizmoManager.OnGizmoTransformObjectEnd += OnGizmoTransformObjectEnd;
+
+    }
+    private void Start()
+    {
+        advanceFowardInputAction.OnStarted += (o) => { isAdvancingFoward = true; };
+        advanceFowardInputAction.OnFinished += (o) => { isAdvancingFoward = false; };
+
+        advanceBackInputAction.OnStarted += (o) => { isAdvancingBackward = true; };
+        advanceBackInputAction.OnFinished += (o) => { isAdvancingBackward = false; };
+
+        advanceLeftInputAction.OnStarted += (o) => { isAdvancingLeft = true; };
+        advanceLeftInputAction.OnFinished += (o) => { isAdvancingLeft = false; };
+
+        advanceRightInputAction.OnStarted += (o) => { isAdvancingRight = true; };
+        advanceRightInputAction.OnFinished += (o) => { isAdvancingRight = false; };
+
+        advanceUpInputAction.OnStarted += (o) => { isAdvancingUp = true; };
+        advanceUpInputAction.OnFinished += (o) => { isAdvancingUp = false; };
+
+        advanceDownInputAction.OnStarted += (o) => { isAdvancingDown = true; };
+        advanceDownInputAction.OnFinished += (o) => { isAdvancingDown = false; };
+    }
+
+    private void Update()
+    {
+
+        if (isAdvancingFoward)
+        {
+            transform.position += transform.forward * keyboardMovementSpeed * Time.deltaTime;
+        }
+        if (isAdvancingBackward)
+        {
+            transform.position += -transform.forward *keyboardMovementSpeed* Time.deltaTime;
+        }
+        if (isAdvancingRight)
+        {
+            transform.position += transform.right * keyboardMovementSpeed * Time.deltaTime;
+        }
+        if (isAdvancingLeft)
+        {
+            transform.position += -transform.right * keyboardMovementSpeed * Time.deltaTime;
+        }
+        if (isAdvancingUp)
+        {
+            transform.position += transform.up * keyboardMovementSpeed * Time.deltaTime;
+        }
+        else if (isAdvancingDown)
+        {
+            transform.position += -transform.up * keyboardMovementSpeed * Time.deltaTime;
+        }
```

Comment:
Here I suggest to use an intermediate velocity `Vector3` to store the direction, and then applying the position change at the end. This would increase maintability by abiding to the DRY principle.

e.g.:
```
Vector3 velocity = Vector3.zero;

if (isAdvancingRight)
    velocity += transform.right;  

...

transform.position += velocity * keyboardMovementSpeed * Time.deltaTime;
```